src/ui/wxWidgets/AddEditPropSheetDlg.cpp

```diff
@@ -1707,19 +1744,57 @@ void AddEditPropSheetDlg::ShowPassword()
   m_IsPasswordHidden = false;
   UpdatePasswordTextCtrl(m_BasicSizer, m_BasicPasswordTextCtrl, m_Password.c_str(), m_BasicUsernameTextCtrl, 0);
   // Disable confirmation Ctrl, as the user can see the password entered
-  ApplyFontPreference(m_BasicPasswordConfirmationTextCtrl, PWSprefs::StringPrefs::PasswordFont);
-  m_BasicPasswordConfirmationTextCtrl->Clear();
+  m_BasicPasswordConfirmationTextCtrl->ChangeValue(wxEmptyString); // Use of ChangeValue instead of SetValue to not trigger an input event.
   m_BasicPasswordConfirmationTextCtrl->Enable(false);
+  m_BasicPasswordConfirmationTextCtrl->SetModified(false);         // Reset of modification flag for password confirmation icon handling.
+  m_BasicShowHideCtrl->SetBitmapLabel(wxUtilities::GetBitmapResource(wxT("graphics/eye_close.xpm")));
+  m_BasicShowHideCtrl->SetToolTip(_("Hide password"));
 }
 
 void AddEditPropSheetDlg::HidePassword()
 {
   m_IsPasswordHidden = true;
   const wxString pwd = m_Password.c_str();
   UpdatePasswordTextCtrl(m_BasicSizer, m_BasicPasswordTextCtrl, pwd, m_BasicUsernameTextCtrl, wxTE_PASSWORD);
-  ApplyFontPreference(m_BasicPasswordConfirmationTextCtrl, PWSprefs::StringPrefs::PasswordFont);
-  m_BasicPasswordConfirmationTextCtrl->ChangeValue(pwd);
+  m_BasicPasswordConfirmationTextCtrl->ChangeValue(pwd);           // Use of ChangeValue instead of SetValue to not trigger an input event.
   m_BasicPasswordConfirmationTextCtrl->Enable(true);
+  m_BasicPasswordConfirmationTextCtrl->SetModified(false);         // Reset of modification flag for password confirmation icon handling.
+  m_BasicShowHideCtrl->SetBitmapLabel(wxUtilities::GetBitmapResource(wxT("graphics/eye.xpm")));
+  m_BasicShowHideCtrl->SetToolTip(_("Show password"));
+}
+
+void AddEditPropSheetDlg::UpdatePasswordConfirmationIcon(int controlId)
+{
+  auto updateLayout = false;
+
+  // There is nothing to do if there is no user input, but the content of
+  // the password input fields may have been changed by the hide/show functionality.
+  if (!m_BasicPasswordTextCtrl->IsModified() && !m_BasicPasswordConfirmationTextCtrl->IsModified()) {
+    return;
+  }
+  // If both passwords entered are the same, the green checkmark icons will appear to indicate the match.
+  if (m_BasicPasswordTextCtrl->GetValue() == m_BasicPasswordConfirmationTextCtrl->GetValue()) {
+    m_BasicPasswordBitmap->SetBitmap(wxUtilities::GetBitmapResource(wxT("graphics/checkmark_green.xpm")));
+    m_BasicPasswordConfirmationBitmap->SetBitmap(wxUtilities::GetBitmapResource(wxT("graphics/checkmark_green.xpm")));
```

Comment:
Not sure how expensive the `GetBitmapResource` call is, but in any case could we load the resource once, assign the result to a variable and use it for both bitmaps?

Reply:
Changed as proposed.